common/src/main/java/io/netty/util/internal/CleanerJava9.java

```diff
@@ -30,46 +32,53 @@
 final class CleanerJava9 implements Cleaner {
     private static final InternalLogger logger = InternalLoggerFactory.getInstance(CleanerJava9.class);
 
-    private static final Method INVOKE_CLEANER;
+    private static final MethodHandle INVOKE_CLEANER_HANDLE;
 
     static {
-        final Method method;
+        final MethodHandle invokeCleanerHandle;
         final Throwable error;
         if (PlatformDependent0.hasUnsafe()) {
             final ByteBuffer buffer = ByteBuffer.allocateDirect(1);
-            Object maybeInvokeMethod = AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
+            Object maybeInvokeMethodHandle = AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
                 try {
+                    MethodHandles.Lookup lookup = MethodHandles.lookup();
                     // See https://bugs.openjdk.java.net/browse/JDK-8171377
-                    Method m = PlatformDependent0.UNSAFE.getClass().getDeclaredMethod(
-                            "invokeCleaner", ByteBuffer.class);
-                    m.invoke(PlatformDependent0.UNSAFE, buffer);
+                    MethodHandle m = lookup.findVirtual(
+                            PlatformDependent0.UNSAFE.getClass(),
+                            "invokeCleaner",
+                            MethodType.methodType(void.class).appendParameterTypes(ByteBuffer.class)).
```

Comment:
`MethodType.methodType(void.class, ByteBuffer.class)`

?

Reply:
yes... was not aware there is such an overload :D